### PR TITLE
TEST: enabled different dataframe testing for linear models: `ElasticNet`, `Lasso`, `Ridge`

### PR DESCRIPTION
**Pull request by samir-nasibli**
### Description 
Enabled different dataframe testing for linear models: `ElasticNet`, `Lasso`, `Ridge`

There is usm_ndarrays introp for these estimators in sklearnex, that is was poorly tested.
In case of fails fix on #1832 , where usm_ndarray_support is updated.

### TODO
will disable GPU testing in case of issues

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.
- [x] I have resolved any merge conflicts that might occur with the base branch.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [x] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

**Review by icfaust — COMMENTED** [review]
please modify to have a comment describing that while these estimators will take a gpu queue, they will run on cpu. this will give information to whoever looks at these next. no todo necessary. otherwise good to go. please confer with #1843 whether ridge modifications are necessary

**Review by ethanglaser — APPROVED** [review]

**Comment by samir-nasibli** [comment]
> please modify to have a comment describing that while these estimators will take a gpu queue, they will run on cpu. this will give information to whoever looks at these next. no todo necessary. otherwise good to go

Thank you @icfaust !
I like the idea to making our code more readable.
Generally I agree that we should comment out such cases, but it is better to do this for such comments inside of the estimators code. I will try to address this proposal.

I think the comments in the test suits will be outdate in some time, it is not obvious for the developer update comments there as well. 
Generally, test suits should work with functionality as black boxes. Comments should be related only tests it self. 

In addition, we have [sklearnex_verbose](https://github.com/intel/scikit-learn-intelex/blob/772fb9b80cb0232d9336af5f51f1b4e4c66be1fd/doc/sources/verbose.rst#L32) which should help the user understand which backend is being used and on which device. I'll see how this can be used in tests, I'll also share

**Comment by samir-nasibli** [comment]
> > please modify to have a comment describing that while these estimators will take a gpu queue, they will run on cpu. this will give information to whoever looks at these next. no todo necessary. otherwise good to go
> 
> Thank you @icfaust ! I like the idea to making our code more readable. Generally I agree that we should comment out such cases, but it is better to do this for such comments inside of the estimators code. I will try to address this proposal.
> 
> I think the comments in the test suits will be outdate in some time, it is not obvious for the developer update comments there as well. Generally, test suits should work with functionality as black boxes. Comments should be related only tests it self.
> 
> In addition, we have [sklearnex_verbose](https://github.com/intel/scikit-learn-intelex/blob/772fb9b80cb0232d9336af5f51f1b4e4c66be1fd/doc/sources/verbose.rst#L32) which should help the user understand which backend is being used and on which device. I'll see how this can be used in tests, I'll also share

In this PR I think I can add comments here. It won't be superfluous

**Comment by samir-nasibli** [comment]
> In addition, we have [sklearnex_verbose](https://github.com/intel/scikit-learn-intelex/blob/772fb9b80cb0232d9336af5f51f1b4e4c66be1fd/doc/sources/verbose.rst#L32) which should help the user understand which backend is being used and on which device. I'll see how this can be used in tests, I'll also share

It seems sklearnex verbose may be helpful here.
```
> SKLEARNEX_VERBOSE=INFO python -m pytest --log-cli-level INFO --pyargs sklearnex.linear_model.tests.test_linear::test_sklearnex_import_lasso[dpctl-SyclQueue_GPU]
============================================================================== test session starts ======================
platform linux -- Python 3.10.13
collected 1 item

::test_sklearnex_import_lasso[dpctl-SyclQueue_GPU] 
--------------------------------------------------------------------------------- live log call -------------------------
INFO     sklearnex:_utils.py:256 sklearn.linear_model.Lasso.fit: running accelerated version on CPU
PASSED                                                                                                              [100%]
```

In anyway I will add comments for the linear models with daal4py backend in certain modules.

File: sklearnex/linear_model/coordinate_descent.py [hunk 1]
```diff
@@ -17,10 +17,14 @@
...
 from onedal._device_offload import support_usm_ndarray
 
+# Note: `sklearnex.linear_model.ElasticNet` only has functional
+# sycl GPU support. No GPU device will be offload.
 ElasticNet.fit = support_usm_ndarray(queue_param=False)(ElasticNet.fit)
 ElasticNet.predict = support_usm_ndarray(queue_param=False)(ElasticNet.predict)
 ElasticNet.score = support_usm_ndarray(queue_param=False)(ElasticNet.score)
 
+# Note: `sklearnex.linear_model.Lasso` only has functional
+# sycl GPU support. No GPU device will be offload.
```
Comment (ethanglaser) [inline sklearnex/linear_model/coordinate_descent.py:27]: is it a concern that we will be removing sycl GPU interfaces?

**Review by ethanglaser — APPROVED** [review]
Assuming green CI

File: sklearnex/linear_model/coordinate_descent.py (hunk 1 above)
Comment (samir-nasibli) [reply sklearnex/linear_model/coordinate_descent.py:27]: Regarding Lasso, Ridge, and ElasticNet, it’s worth noting that there is no GPU offloading for these specific algorithms.
However, with dpct/dpnp and a GPU SYCL queue, you can still use them as input and output, but all algorithm computations will occur on the CPU device, even when a GPU SYCL queue is provided.

File: sklearnex/linear_model/ridge.py [hunk 2]
```diff
@@ -17,6 +17,8 @@
 from daal4py.sklearn.linear_model import Ridge
 from onedal._device_offload import support_usm_ndarray
 
+# Note: `sklearnex.linear_model.Ridge` only has functional
+# sycl GPU support. No GPU device will be offload.
```
Comment (samir-nasibli) [inline sklearnex/linear_model/ridge.py:21]:
```suggestion
# sycl GPU support. No GPU device will be offloaded.
```
typo

**Comment by samir-nasibli** [comment]
Thank you for the review! Internal CI is green.
Going to fix minor comment typo and merge it.
